modules/core/client/services/location.client.service.js

```diff
@@ -65,16 +65,34 @@
      */
     function getBounds(geolocation) {
       if (!geolocation || !geolocation.bbox || !angular.isArray(geolocation.bbox) || geolocation.bbox.length !== 4) {
-        return false;
+        var center = geolocation.center;
+        if (!geolocation || !center || !angular.isArray(center) || center.length !== 2) {
+          return false;
+        } else {
+          var borderFromCenter = .002;
+          return getBoundsObject(
+            parseFloat(center[1]) + borderFromCenter,
+            parseFloat(center[0]) - borderFromCenter,
+            parseFloat(center[1]) - borderFromCenter,
+            parseFloat(center[0]) + borderFromCenter);
+        }
       }
+      return getBoundsObject(
+        parseFloat(geolocation.bbox[3]),
+        parseFloat(geolocation.bbox[2]),
+        parseFloat(geolocation.bbox[1]),
+        parseFloat(geolocation.bbox[0]));
+    }
+
```

Comment:
POIs only have ‘center’, an array containing a point, while cities, countries have a bounding box ‘bbox’ array containing 2 points.

Without the changes above, selecting a POI doesn't do anything because there's no ‘bbox’ array

Reply:
Makes sense, thanks for elaborating!